webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/users/UserRepository.scala

```diff
@@ -860,8 +862,9 @@ class JdbcUserRepository(doobie: Doobie) extends UserRepository {
     transactIOResult(s"Error when retrieving user information")(xa => sql.query[UserInfo].to[List].transact(xa))
   }
 
-  override def get(userId: String): IOResult[Option[UserInfo]] = {
-    val sql = sql"""select * from users where id = ${userId}"""
+  override def get(userId: String, isCaseSensitive: Boolean): IOResult[Option[UserInfo]] = {
+    val filter = if (isCaseSensitive) fr"id = ${userId} " else fr"lower(id) = ${userId.toLowerCase()}"
+    val sql    = fr"""select * from users where""" ++ filter
 
     transactIOResult(s"Error when retrieving user information for '${userId}'")(xa => sql.query[UserInfo].option.transact(xa))
```

Comment:
here, you need to manage the case where there is more than one result

Reply:
That's a well seen edge case !
:white_check_mark: Fixed with https://github.com/Normation/rudder/pull/5655/commits/1c5f6d4bed99d82742587af9372e55afec84d1e8 and https://github.com/Normation/rudder/pull/5655/commits/a325e13d14574a7b3c793c424a1ff497ae00f94f, I made it throw an `Inconsistency` and added test for both in-memory and jdbc repository